src/App.tsx

```diff
@@ -8,55 +8,157 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { getUsers } from './api/users';
+import { User } from './types/User';
+import { ErrorTypes } from './constants';
+import { getPostsByUserId } from './api/posts';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
+import { getPostComments } from './api/comments';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoaderVisible, setIsLoaderVisible] = useState(false);
+  const [isCommentFormOpen, setIsCommentFormOpen] = useState(false);
+  const [isPostlistVisible, setIsPostListVisible] = useState(false);
+  const [isCommentsFetching, setIsCommentsFetching] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [errorType, setErrorType] = useState<ErrorTypes | null>(null);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isCommentsListtHidden, setIsCommentsListHidden] = useState(true);
```

Comment:
move some states to other components for example in my opinion  isCommentFormVisible can be internal state of PostDetails component